tap_googleads/tap.py

```diff
@@ -43,45 +43,70 @@ class TapGoogleAds(Tap):
     # TODO: Add Descriptions
     config_jsonschema = th.PropertiesList(
         th.Property(
-            "oauth_credentials.client_id",
-            th.StringType,
-        ),
-        th.Property(
-            "oauth_credentials.client_secret",
-            th.StringType,
+            "oauth_credentials",
+            th.ObjectType(
+                th.Property(
+                    "client_id",
+                    th.StringType,
+                    required=True,
+                ),
+                th.Property(
+                    "client_secret",
+                    th.StringType,
+                    required=True,
+                    secret=True,
+                ),
+                th.Property(
+                    "refresh_token",
+                    th.StringType,
+                    required=True,
+                    secret=True,
+                ),
+                additional_properties=False,
+            ),
+            required=True,
```

Comment:
Pretty niche edge case, but requiring these sub-properties breaks the proxy auth flow.

```suggestion
            "oauth_credentials",
            th.OneOf(
                th.ObjectType(
                    th.Property(
                        "client_id",
                        th.StringType,
                        required=True,
                    ),
                    th.Property(
                        "client_secret",
                        th.StringType,
                        required=True,
                        secret=True,
                    ),
                    th.Property(
                        "refresh_token",
                        th.StringType,
                        required=True,
                        secret=True,
                    ),
                    additional_properties=False,
                ),
                th.ObjectType(
                    th.Property(
                        "refresh_proxy_url",
                        th.StringType,
                        required=True,
                    ),
                    th.Property(
                        "refresh_proxy_url_auth",
                        th.StringType,
                        secret=True,
                    ),
                    additional_properties=False,
                ),
            ),
            required=True,
```

Reply:
Following up here, since this change was not quite right:

20392b6e8b08cae6d06ab3f33e37360368c0f386
f83321f552b38fc6c726e3cd9314556188ccddc5